doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+the platform, as it contains headers and tools that aren't included in the FOSS
+versions found in ``brew``.
+
+.. code:: sh
+
+   xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+.. code:: sh
+
+   brew install meson cmake pkg-config docutils imagemagick pygments
+
+The following dependencies are technically optional, and can be pulled in at
+build-time, but they will reduce compile times during development, so it's
+recommended to install them.
+
+.. code:: sh
+
+   brew install freetype2 libzip opusfile libvorbis webp sdl2
+
+Optionally, if you're on macOS and compiling for macOS, you can to install
+`create-dmg <https://github.com/create-dmg/create-dmg>`__, which will allow
+you to have nicer-looking macOS ``.dmg`` files for distribution:
+
+.. code:: sh
+
+   brew install create-dmg
+
+As of 2021-08-05, you should **not** install the following packages via
+Homebrew, as the versions available do not compile against Taisei correctly.
+If you're having mysterious errors, ensure that they're not installed.
+
+-  ``spirv-tools``
+-  ``spirv-cross``
+-  ``sdl2_mixer``
+
+.. code:: sh
+
+   brew remove spirv-tools spirv-cross sdl2_mixer
+
+In addition, if you're trying to compile on an older version of macOS
+(i.e: <10.12), SDL2 may not compile correctly on Homebrew (as of 2019-02-19).
+Let ``meson`` pull in the corrected version for you via subprojects.
+
+**NOTE:** While Homebrew's optional dependencies greatly improve compile times,
+if you can't remove packages that give you errors from your system for whatever
+reason, you can force ``meson`` to use its built-in subprojects by using
+``--wrap-mode`` (more on that later).
+
+Windows
+'''''''
+
+Taisei uses `mstorsjo/llvm-mingw <https://github.com/mstorsjo/llvm-mingw>`__ to
+achieve cross-compiling on Windows. Cross-compiling for Windows ends up being
+easier to maintain and more consistent than attempting to use Microsoft's native
+toolchain.
+
+On Linux, you'll need the following tools for cross-compiling Taisei for Windows
+on Linux:
+
+- ``llvm-mingw``
+- `nsis <https://nsis.sourceforge.io/Main_Page>`__ >= 3.0
+
+On macOS, you're probably better off using Docker and the
+`Docker container <https://hub.docker.com/r/mstorsjo/llvm-mingw/>`__ that
+``llvm-mingw`` provides, and installing ``nsis`` on top of it.
+
+Another options for Windows-based computers is leveraging Windows
+10's
+`Windows For Linux (WSL) Subsystem <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__
+to cross-compile to Windows using their Ubuntu image.
+
+Checking Out Code
+-----------------
+
+First, you'll need to checkout the repository. You can do that with the
+following:
+
+.. code:: sh
+
+   git clone https://github.com/taisei-project/taisei.git
+   cd taisei/
+   git submodule update --init --recursive
```

Comment:
This can be done in one command `git clone --recurse-submodules https://github.com/taisei-project/taisei.git`